File: docs/src/testers/agents/config/bp_vision.py
# @sniptest filename=bp_vision.py
# @sniptest show=5-9
from notte_sdk import NotteClient

client = NotteClient()
with client.Session() as session:
    # Text-only site
    text_agent = client.Agent(session=session, use_vision=False)

    # Image-heavy site
    agent = client.Agent(session=session, use_vision=True)
    status = session.status()
    assert text_agent.request.use_vision is False
    assert text_agent.request.session_id == status.session_id
    assert agent.request.use_vision is True
    assert agent.request.session_id == status.session_id

File: docs/src/testers/agents/config/bp_vision.ts
// @sniptest filename=bp_vision.ts
// @sniptest show=7-11
import assert from 'node:assert/strict';
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

const status = await client.Session().use(async session => {
  // Text-only site
  let agent = client.Agent({ session, use_vision: false });

  // Image-heavy site
  agent = client.Agent({ session, use_vision: true });
  const status = await session.status();
  assert.equal(agent['request']?.use_vision, true);
  assert.equal(agent['request']?.session_id, status.session_id);
  return status;
});

export { status };
